### ISSUE TITLE
--backend=xcode fails to generate a project file

### ISSUE THREAD

**Issue by paxos1977** [mesonbuild/meson#337]
Trying to generate an Xcode project from a trivial example, and I'm getting the following errors:

    MyMachine:build user_name$ meson .. --backend=xcode
    The Meson build system
    Version: 0.27.0
    Source dir: /Volumes/user_name/proj/_work/test_meson
    Build dir: /Volumes/user_name/proj/_work/test_meson/build
    Build type: native build
    Build machine cpu family: x86_64
    Build machine cpu: x86_64
    Project name: test meson
    Native cpp compiler: c++ (clang 7.0.0)
    Message: Configuring clang-7.0
    Build targets in project: 1
    Traceback (most recent call last):
      File "/Volumes/user_name/bin/meson-0.27.0/meson.py", line 208, in run
        app.generate()
      File "/Volumes/user_name/bin/meson-0.27.0/meson.py", line 161, in generate
        g.generate(intr)
      File "/Volumes/user_name/bin/meson-0.27.0/xcodebackend.py", line 66, in generate
        self.serialise_tests()
      File "/Volumes/user_name/bin/meson-0.27.0/backends.py", line 128, in serialise_tests
        self.write_test_file(datafile)
      File "/Volumes/user_name/bin/meson-0.27.0/backends.py", line 279, in write_test_file
        fname = [os.path.join(self.environment.get_build_dir(), self.get_target_filename(t.get_exe()))]
      File "/Volumes/user_name/bin/meson-0.27.0/backends.py", line 63, in get_target_filename
        targetdir = self.get_target_dir(target)
      File "/Volumes/user_name/bin/meson-0.27.0/xcodebackend.py", line 55, in get_target_dir
        dirname = os.path.join(target.get_subdir(), self.environment.coredata.buildtype)
    AttributeError: 'CoreData' object has no attribute 'build type'

The last line is the error. I have no idea what this means. Everything builds fine with ninja. 

BTW, I saw mention in another thread what you think about Xcode's project format ;) I have some limited experience working with it, if the generator needs some loving let me know and I'll try to dig in when I have time.

**Comment by jpakkane** [comment]
I fixed this issue but it still does not work fully. One issue is that xcode seems to have weird limitations, such that you can't have the @ character in strings (this is used as a separator between different target types). If you manually fix those it will fail with this helpful error message:

    2015-12-24 23:31:37.734 xcodebuild[23030:1358522] The data couldn’t be read because it isn’t in the correct format.
    xcodebuild: error: Unable to read project 'trivial test.xcodeproj'.
	Reason: Project /Users/jpakkane/meson/work area/trivial test.xcodeproj cannot be opened because the project file cannot be parsed.

Any help is appreciated.

**Comment by Rogiel** [comment]
I am trying to track this issue down, so far I have found 3 issues:

1. Missing `};` for closing **objects**
2. Another missing `}` (without semicolon) to close the main project
3. A missing `rootObject` pointing towards the root object (I have tried to find a valid root, but failed to do so)

After fixes those two issues, Xcode prints the following on the console: 
````
2/24/16 10:31:20.528 PM Xcode[61342]: The project contains no build configurations - it may have been damaged
````

This could be due to using the wrong root. By inspecting a valid project it seems that the root points to a `PBXProject` instance.
